src/Analyzers/VisualBasic/CodeFixes/IncludeParenthesesForMethodInvocations/VisualBasicIncludeParenthesesForMethodInvocationsCodeFixProvider.vb

```diff
@@ -0,0 +1,85 @@
+﻿' Licensed to the .NET Foundation under one or more agreements.
+' The .NET Foundation licenses this file to you under the MIT license.
+' See the LICENSE file in the project root for more information.
+
+Imports System.Collections.Immutable
+Imports System.Composition
+Imports System.Diagnostics.CodeAnalysis
+Imports System.Threading
+Imports Microsoft.CodeAnalysis.CodeActions
+Imports Microsoft.CodeAnalysis.CodeFixes
+Imports Microsoft.CodeAnalysis.Diagnostics
+Imports Microsoft.CodeAnalysis.Editing
+Imports Microsoft.CodeAnalysis.VisualBasic.Syntax
+
+Namespace Microsoft.CodeAnalysis.VisualBasic.IncludeParenthesesForMethodInvocations
+    <ExportCodeFixProvider(LanguageNames.VisualBasic), [Shared]>
+    Friend Class VisualBasicIncludeParenthesesForMethodInvocationsCodeFixProvider
+        Inherits SyntaxEditorBasedCodeFixProvider
+
+        <ImportingConstructor>
+        <SuppressMessage("RoslynDiagnosticsReliability", "RS0033:Importing constructor should be [Obsolete]", Justification:="Used in test code: https://github.com/dotnet/roslyn/issues/42814")>
+        Public Sub New()
+        End Sub
+
+        Public Overrides ReadOnly Property FixableDiagnosticIds As ImmutableArray(Of String) =
+            ImmutableArray.Create(IDEDiagnosticIds.IncludeParenthesesForMethodInvocationsDiagnosticId)
+
+        Friend Overrides ReadOnly Property CodeFixCategory As CodeFixCategory = CodeFixCategory.CodeStyle
+
+        Public Overrides Async Function RegisterCodeFixesAsync(context As CodeFixContext) As Task
+            Dim root = Await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(False)
+            For Each diagnostic In context.Diagnostics
+                Dim node = root.FindNode(diagnostic.AdditionalLocations(0).SourceSpan)
+                If TypeOf node IsNot InvocationExpressionSyntax Then
+                    ' The code flow comes here. There are multiple nodes with the same span.
+                    ' The node I want isn't always the most inner one. Not sure about the correct fix here.
+                    Return
+                End If
+                Dim invocationExpression = DirectCast(node, InvocationExpressionSyntax)
```

Comment:
@CyrusNajmabadi What do you think is the ideal solution here?

Reply:
I've fixed the problem of getting the correct node (There is another problem now that I'm looking into it).
But I don't feel my approach is good enough. There is some kind of duplicates between the RegisterCodeFixAsync and the FixAllAsync methods.